File: post-comments/src/main/java/com/harsha/organize/entity/Comment.java
package com.harsha.organize.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;


@Data
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "comment")
@Entity
public class Comment {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;
    private String name;
    private String email;
    @Column(nullable = false, columnDefinition = "text")
    private String body;
    private LocalDateTime createAt;
    private LocalDateTime updateAt = LocalDateTime.now();
    @ManyToOne
    @JoinColumn(name = "post_id", referencedColumnName = "id")
    private Post post;
}

File: post-comments 2/src/main/kotlin/com/harsha/organize/entity/Comment.kt
package com.harsha.organize.entity

import org.springframework.data.annotation.Id
import org.springframework.data.relational.core.mapping.Table
import java.time.LocalDateTime


@Table(name = "comment")
data class Comment(
    @Id
    var id: Int,
    var name: String = "",
    var email: String = "",
    var body: String = "",
    var createAt: LocalDateTime,
    var updateAt: LocalDateTime = LocalDateTime.now()
)

